src/wolff_producer.erl

```diff
@@ -376,13 +383,20 @@ send_to_kafka(#{sent_reqs := Sent,
     replayq:pop(Q, #{bytes_limit => BytesLimit, count_limit => 999999999}),
   {FlatBatch, Calls} = get_flat_batch(Items, [], []),
   [_ | _] = FlatBatch, %% assert
+  BatchSize = erlang:length(FlatBatch),
+  wolff_metrics:queuing_change(Config, -BatchSize),
+  wolff_metrics:inflight_change(Config, BatchSize),
```

Comment:
If this worker crashes, we might need to run change those metrics in `terminate`?

Reply:
Currently, if the following timeline happens:

1. A client sends a message.
2. The message is sent to Kafka.
3. Before an ack is received from Kafka, the wolff producer crashes in a non-brutal way (i.e., `exit(Producer, some_error)`).
4. At this point, `inflight` is at +1 forever, and later, when the producer restarts, `queuing` goes to -1 forever (see trace below). 

<details>
<summary> Example trace </summary>

```elixir
[
  {-576460685686716481, {[:wolff, :queuing], %{counter_inc: 1}, %{}}},
  {-576460685686699076, {[:wolff, :queuing], %{counter_inc: -1}, %{}}},
  {-576460685686687752, {[:wolff, :inflight], %{counter_inc: 1}, %{}}},
  ## crash happens here, then producer is restarted
  {-576460682568348225, {[:wolff, :queuing], %{counter_inc: -1}, %{}}},
  {-576460682568304233, {[:wolff, :inflight], %{counter_inc: 1}, %{}}},
  {-576460682565842352, {[:wolff, :success], %{counter_inc: 1}, %{}}},
  {-576460682565770383, {[:wolff, :inflight], %{counter_inc: -1}, %{}}}
]
```

</details>

However, talking to Stone, I understood that we should reset those counters at a higher level (e.g., the client using Wolff) when such a crash would happen.  So we may ignore this issue here.

Reply:
Right, a process may get killed, in which case no `terminate` call anyway.

The only control point seems to be at the higher level.